src/java/org/apache/cassandra/service/CassandraDaemon.java

```diff
@@ -289,6 +290,8 @@ protected void setup()
 
         setupVirtualKeyspaces();
 
+        LegacySystemKeyspaceToNodes.convertToNodesFilesIfNecessary();
```

Comment:
I would need to double-check whether it doesn't introduce any regression in astra 🤞